geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PartitionedRegionClearWithRebalanceDUnitTest.java

```diff
@@ -0,0 +1,376 @@
+          + results.getTotalBucketCreatesCompleted() + results.getTotalPrimaryTransfersCompleted();
+      assertThat(combinedResults, greaterThan(0));
+
+      // Verify that no primary transfers failed during the rebalance
+      assertThat(cacheRule.getCache().getInternalResourceManager().getStats()
+          .getRebalancePrimaryTransfersFailed(), is(0));
+    });
+
+    clearInvocation.await();
+
+    // Assert that the region is empty
+    assertRegionIsEmpty(asList(accessor, server1, server2));
+  }
+
+  @Test
+  @Parameters(method = "vmsAndRegionTypes")
+  @TestCaseName("[{index}] {method}(ClearCoordinator:{0}, RebalanceCoordinator:{1}, RegionType:{2})")
+  public void clearRegionDuringRebalanceClearsRegionWhenNonCoordinatorIsBounced(
+      TestVM clearCoordinatorVM, TestVM rebalanceVM, RegionShortcut regionType)
+      throws InterruptedException {
+    AsyncInvocation<?> clearInvocation = setupAndPrepareClear(clearCoordinatorVM, regionType);
+
+    // Server 2 is never the clear coordinator
+    registerVMKillerAsCacheWriter(Collections.singletonList(server2));
+
+    getVM(rebalanceVM.vmNumber).invoke(() -> {
+      // Start a rebalance and wait until bucket creation for redundancy recovery has started before
+      // signalling the blackboard
+      RebalanceResults results = startRebalanceAndGetResults();
+
+      // Verify that rebalance did some work
+      int combinedResults = results.getTotalBucketTransfersCompleted()
+          + results.getTotalBucketCreatesCompleted() + results.getTotalPrimaryTransfersCompleted();
+      assertThat(combinedResults, greaterThan(0));
+    });
+
+    clearInvocation.await();
+
+    // Bring server 2 back online and assign buckets
+    server2.invoke(() -> {
+      cacheRule.createCache();
+      initDataStore(regionType);
+      await().untilAsserted(
+          () -> Assertions.assertThat(InternalDistributedSystem.getConnectedInstance())
+              .isNotNull());
+      PartitionRegionHelper.assignBucketsToPartitions(cacheRule.getCache().getRegion(REGION_NAME));
+    });
+
+    // Assert that the region is empty
+    assertRegionIsEmpty(asList(accessor, server1, server2));
+  }
+
+  /**
+   * Shutdowns a member while the clear operation is in progress.
+   * The writer is only installed on the member the test wants to shutdown, doesn't matter whether
+   * it's the clear coordinator or another member holding primary buckets.
+   */
+  public static class MemberKiller extends CacheWriterAdapter<String, String> {
+
+    @Override
+    public synchronized void beforeRegionClear(RegionEvent<String, String> event)
+        throws CacheWriterException {
+      InternalDistributedSystem.getConnectedInstance().stopReconnectingNoDisconnect();
+      MembershipManagerHelper.crashDistributedSystem(
+          InternalDistributedSystem.getConnectedInstance());
+      await().untilAsserted(
+          () -> Assertions.assertThat(InternalDistributedSystem.getConnectedInstance()).isNull());
```

Comment:
Do we need additional check to see cache is closed...When this vm is restarted, it creates the cache again.

Reply:
I can add a check.